kyo-core/shared/src/main/scala/kyo/Channel.scala

```diff
@@ -525,17 +525,17 @@ object Channel:
                         case Absent =>
                             Absent
                         case Present(Put.Value(value, promise)) =>
-                            discard(promise.complete(Result.unit))
+                            discard(promise.complete(Result.succeed(())))
```

Comment:
this is something not great about this new encoding of `Fiber`.  The completion now takes a `Result[E, A < S]` so `succeed` is necessary to lift `()` to `Unit < Any`

Reply:
```scala
        extension [S](self: Promise[Unit, S])
            def succeed(using Frame): Boolean < Sync = succeedWith(Kyo.unit)

            def succeedWith(using Frame)(v: Unit < S): Boolean < Sync =
                Sync.Unsafe(Unsafe.complete(self)(Result.succeed(v)))
        end extension

        extension [A, S](self: Promise[A, S])(using NotGiven[S <:< Abort[Any]])

            def succeedWith(v: A < S)(using Frame): Boolean < Sync = self.complete(Result.succeed(v))

Reply:
I've added `completeUnit`. We use the `With` suffix to indicate that the method takes a continuation function